src/mmw/apps/modeling/views.py

```diff
@@ -373,7 +382,11 @@ def get_list_of_workers():
 
     workers = filter(predicate,
                      get_list_of_workers())
-    return random.choice(workers)
+    return workers
+
+
+def choose_worker():
```

Comment:
Correct me if I misunderstand your point, but within the context of `choose_worker`'s use I think the list of available workers will always be up-to-date at the time of invocation since it's just a thin selector over `get_living_workers`, unless the python interpreter does some type of caching or memoization internally. From the perspective of all consuming methods, I *think* the change to `choose_worker` is invisible; I split `get_living_workers` into a separate declaration so that I could bind the result for re-use specifically in `_initiate_mapshed_job` and leave every other use unaffected.

To your question, pinging celery for each job was a bottleneck with noticeable effects, especially for job chains with more than a couple jobs and where the jobs themselves are not especially long-running. You're correct that the list of workers in the mapshed chain could grow stale, but the chain itself isn't very long running, and qualitatively I think there's some confidence in the stability of the workers in the +10 second time frame.

I also created #1535 to address the concern of leaving changes in celery workers unhandled.